### PR TITLE
refactor(core): use html whitespace instead shifted value

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [x] Refactoring
-   [ ] Code style update
-   [ ] Build or CI related changes
-   [ ] Documentation content changes

## What is the current behavior?

Need changes before upgrade to https://github.com/TinkoffCreditSystems/taiga-ui/pull/988

## What is the new behavior?

- with value and with filler and with postfix

image

- without value and with filler and with postfix

image

- without value and without filler and with postfix

image

- with value and without filler and with postfix

image

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Comment by splincode** [comment]
@waterplea @nsbarsukov @vladimirpotekhin  please, review me))

**Comment by waterplea** [comment]
> @waterplea @nsbarsukov @vladimirpotekhin please, review me))

There's no point until RTL PR gets merged. And I just noticed InputSlider/InputRange also need to support RTL so I need to add it first.

File: projects/core/components/primitive-textfield/primitive-textfield.component.ts [hunk 1]
```diff
@@ -251,7 +251,7 @@ export class TuiPrimitiveTextfieldComponent
     get computedFiller(): string {
         return this.hasExampleText
             ? this.controller.exampleText
-            : this.filler.slice(this.value.length);
+            : this.filler?.toString().slice(this.value?.toString().length);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/primitive-textfield/primitive-textfield.component.ts:254]: @waterplea 

https://taiga-ui.dev/components/primitive-textfield/API?tuiMode=null&filler=123&value=123&pseudoFocused=true

image

[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/components/primitive-textfield/primitive-textfield.component.ts:254]: It gets incorrectly parsed and passed as a number which is wrong type. You should rather fix documentation parsing URL params. I believe it also parses "false" as a string "false" making it truthy.

image

[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/primitive-textfield/primitive-textfield.component.ts:254]: But why can't we make improvements in this place? Thus, we will protect the runtime behavior.

[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/components/primitive-textfield/primitive-textfield.component.ts:254]: We do not protect people against using incorrect types. Taiga was always strictly typed, demanding discipline from developers as one of its requirements and it's one of the reasons devs liked it internally 🙂

**Pull request closed without merging**